### PR TITLE
Change NoScope to fire a hilariously overpowered homing arrow

### PR DESCRIPTION
**Pull request by solunareclipse1**
Also contains some other changes that I had to do to get the project running in my IDE. I pretty much only mod on Forge so there's a good chance I just don't know what I'm doing; I will revert such changes upon request if they are problematic.

I ported this feature from a Forge mod I've been working on for a little while now, manually translating it from Mojang maps to Yarn. I might have messed it up in a couple spots but everything *seems* to be working fine.

Contains some voice clips from the Sniper from Team Fortress 2, but I don't think this will cause any copyright issues. People make TF2 animations using the voice clips all the time, and they are publicly avaliable for download on the Team Fortress 2 wiki. If copyright is an issue it's quite simple to just remove the sounds, they arent necessary (although it would become a lot less funny).

Some things about the arrow:
- It pathfinds around obstacles, borrowing much from Minecraft's pathfinding logic (A*)
- It also gives up relatively easily if it cant find a path (and stops pathfinding)
- Only lasts for 10 seconds, after which it vanishes
- Is not saved to the disk to prevent weird stuff
- Re-implements a lot of the Entity and Projectile tick logic with some modifications
- Target for the arrow can be manually set by right-clicking while the arrow is active
  - As a result, only 1 Arrow can be out at a time per NoScope item
- Arrow deals damage in the same way as the /kill command
- Arrow can be seen through walls (glowing effect) to better keep track of it
- Arrow is really fun to watch do its thing
- I tried to strip things down, but there may still be some remnants from the aforementioned mod I've been working on

Project successfully builds in my IDE, and everything I changed seems to be working as I intend, both in dev and in production. I have a video file of the arrow in action but it's a bit too big for GitHub, let me know if you want me to upload that somewhere and put it here.

### CODE REVIEW

**Comment by ENDERZOMBI102** [comment]
how did i not get a notification for this...
i'll check it out and see if any changes are needed, else i'll merge in later today

**Comment by ENDERZOMBI102** [comment]
if you are willing tho apply/comment on those changes, please do, else I'll just merge

**Comment by solunareclipse1** [comment]
ill have a look, assuming i can get the project up and running again

File: src/main/java/fr/anatom3000/gwwhit/GWWHITClient.java [hunk 1]
```diff
@@ -20,15 +25,21 @@ public class GWWHITClient implements ClientModInitializer {
...
         EventListeners.registerClient();
         Commands.registerClient();
         Registry.register(Registry.SOUND_EVENT, WHISTLE_SOUND, WHISTLE_SOUND_EVENT);
         Registry.register(Registry.SOUND_EVENT, MOJAAAANG_SOUND, MOJAAAANG_SOUND_EVENT);
         Registry.register(Registry.SOUND_EVENT, ROCK_SOUND, ROCK_SOUND_EVENT);
+        // server needs to know about these, unless somebody wants to fuck with the SentientArrow code
+        Registry.register(Registry.SOUND_EVENT, GWWHIT.SNIPER_NS, GWWHIT.SNIPER_NS_EVENT); // Good shot mate!
+        Registry.register(Registry.SOUND_EVENT, GWWHIT.SNIPER_THX, GWWHIT.SNIPER_THX_EVENT); // 'preciate it.
 
+        EntityRendererRegistry.INSTANCE.register(EntityRegistry.SENTIENT_ARROW_ENTITY, (ctx) -> new SentientArrowRenderer(ctx));
```
[the remark this answers is not in the document]
Comment (solunareclipse1) [reply src/main/java/fr/anatom3000/gwwhit/GWWHITClient.java:41]: done

File: build.gradle [hunk 2]
```diff
@@ -20,7 +20,15 @@ repositories {
 	maven { url 'https://maven.terraformersmc.com/releases/' }
 	maven { url 'https://maven.shedaniel.me/' }
     maven {	url 'https://repo.repsy.io/mvn/enderzombi102/mc/' }
-    maven { url 'https://storage.googleapis.com/devan-maven/' }
+    //maven { url 'https://storage.googleapis.com/devan-maven/' } // seems to be an old repository for AARP? it doesnt exist anymore
+    //maven { url 'https://ueaj.dev/maven' } // this is apparently where AARP is supposed to reside now, but at the time of writing (Oct 7 2023) its https is broken and therefore doesnt work
+    // because of the things outlined above, ive had to resort to using curseforge maven, and a different version of AARP. my apologies
```
[the remark this answers is not in the document]
Comment (solunareclipse1) [reply build.gradle:25]: seems like there isnt really anything for me to do here, then?

File: resources.gradle [hunk 3]
```diff
@@ -8,7 +8,9 @@ new XmlParser().parse("https://knowyourmeme.com/memes.rss").channel.item.each {
 }
 project.ext.codename = possibleNames.get(new Random(project.mod_version.hashCode()).nextInt(possibleNames.size()))
 System.out.println("Chose codename \"" + project.ext.codename + "\" from " + possibleNames.size() + " choices")
-/*
+
+// im not really sure why this was commented out, but the game refused to launch without downloading it because somebody made beemovie.txt loadbearing
```
[the remark this answers is not in the document]
Comment (solunareclipse1) [reply resources.gradle:12]: done

**Comment by solunareclipse1** [comment]
@ENDERZOMBI102 dealt with everything besides the maven repo, as i dont think thats really fixable on my end. game still boots. anything else?

**Comment by ENDERZOMBI102** [comment]
> @ENDERZOMBI102 dealt with everything besides the maven repo, as i dont think thats really fixable on my end. game still boots. anything else?

Maybe add yourself to the list of contributors? if you want ofc.
Apart this, looks amazing and ready to be merged!

**Pull request merged**